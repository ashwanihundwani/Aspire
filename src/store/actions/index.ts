// export action creators

import * as debitCardActions from './debitCardActions';


export const ActionCreators = Object.assign(
  {},
  debitCardActions
);
