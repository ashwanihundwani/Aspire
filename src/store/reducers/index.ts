import { combineReducers } from 'redux';

import debitCardReducer from './debitCardReducer';
import loadingReducer from './loadingReducer';


export default combineReducers({
    debitCardReducer,
    loadingReducer
});