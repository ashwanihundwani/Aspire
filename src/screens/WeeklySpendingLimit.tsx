import React, { useState } from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity } from 'react-native';
import { AppColors, layouting, WeeklySpendingString, testIDs } from '../utils/constants';
import { avenirFont, FontStyle, windowHeight, windowWidth } from '../utils/utils';
import Images from '../../src/assets/images';
import CurrencyWithAmount from '../components/CurrencyWithAmount';
import AmountButton from '../components/AmountButton';
import { DebitCardResponse, DebitCardRequest } from '../models/DebitCardModel';
import { useSelector, useDispatch } from 'react-redux';
import { updateDebitCard } from '../store/actions/debitCardActions';

const amounts: string[] = ["5,000", "10,000", "20,000"]
const spacingFactor = layouting.spacingFactor
const screenWidth = windowWidth()
const bottomSheetHeight = windowHeight() - 130 // Calculation based on layouts in design
const separatorWidth = screenWidth - (spacingFactor * 2)
const buttonContainerWidth = separatorWidth
const saveButtonSpacing = 0.135 * screenWidth
const saveButtonWidth = screenWidth - (saveButtonSpacing * 2)

const WeeklySpendingLimit = (props: any) => {

  const dispatch = useDispatch();
  const debitCardResponse: DebitCardResponse = useSelector((state: any): DebitCardResponse => state.debitCardReducer.debitCard)

  const [selectedAmount, setSelectedAmount] = useState(debitCardResponse ? debitCardResponse.weeklySpendLimit : "")

  const amountButtonClicked = (index: number) => {
    setSelectedAmount(amounts[index])
  }

  return (
    <View style={styles.container}>
      <Text style={styles.headerTitle}>{WeeklySpendingString.headerTitle}</Text>
      <View style={styles.bottomSheet}>
        <View style={styles.bottomSheetContentContainer}>
          <View style={styles.descriptionContainer}>
            <Image style={styles.descriptionIcon} source={Images.weeklySpent.spent} />
            <Text>{WeeklySpendingString.descriptionMessage}</Text>
          </View>
          <CurrencyWithAmount amount={selectedAmount} textColor={AppColors.descriptionColor} />
          <View style={styles.separator} />
          <Text style={styles.infoText}>{WeeklySpendingString.infoMessage}</Text>
          <View style={styles.buttonsContainer}>
            <AmountButton amount={amounts[0]} onClick={() => {
              amountButtonClicked(0)
            }} />
            <AmountButton amount={amounts[1]} onClick={() => {
              amountButtonClicked(1)
            }} />
            <AmountButton amount={amounts[2]} onClick={() => {
              amountButtonClicked(2)
            }} />
          </View>
        </View>
        <View style={styles.saveButtonContainer}>
          <TouchableOpacity
            disabled={selectedAmount.length == 0}
            onPress={() => {
              let debitCardRequest: DebitCardRequest = { ...debitCardResponse }
              debitCardRequest.weeklySpendLimit = selectedAmount
              dispatch(updateDebitCard(debitCardRequest))
              props.navigation.goBack()
            }}
            style={[styles.saveButton, selectedAmount.length > 0 ? { backgroundColor: AppColors.lightGreen } : { backgroundColor: AppColors.disabledSaveButton }]}>
            <Text style={styles.saveButtonText}>{WeeklySpendingString.saveButton}</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: AppColors.navigationBackColor
  },
  headerTitle: {
    marginTop: 10,
    fontFamily: avenirFont(FontStyle.Bold),
    fontSize: 24,
    color: AppColors.navigationTintColor,
    marginLeft: layouting.spacingFactor
  },
  bottomSheet: {
    flex: 1,
    width: "100%",
    height: bottomSheetHeight,
    marginTop: 60,
    backgroundColor: AppColors.whiteColor,
    borderTopLeftRadius: 24,
    borderTopRightRadius: 24,
  },
  bottomSheetContentContainer: {
    flex: 1
  },
  descriptionContainer: {
    flexDirection: "row",
    marginLeft: layouting.spacingFactor,
    marginTop: layouting.spacingFactor,


  },
  descriptionIcon: {
    marginRight: 12
  },
  descriptionText: {
    color: AppColors.descriptionColor,
    fontSize: 14,
    fontFamily: avenirFont(FontStyle.Medium)
  },
  separator: {
    marginLeft: spacingFactor,
    width: separatorWidth,
    height: 0.7,
    marginTop: 3,
    backgroundColor: AppColors.separatorColor
  },
  infoText: {
    marginTop: 14,
    fontSize: 13,
    fontFamily: avenirFont(FontStyle.Regular),
    color: AppColors.darkTextColor,
    marginLeft: spacingFactor
  },
  buttonsContainer: {
    marginTop: spacingFactor,
    width: buttonContainerWidth,
    marginLeft: spacingFactor,
    flexDirection: "row",
    justifyContent: "space-around"
  },

  saveButtonContainer: {
    justifyContent: "flex-end",
    marginBottom: 50
  },
  saveButton: {
    backgroundColor: AppColors.lightGreen,
    width: saveButtonWidth,
    height: saveButtonSpacing,
    alignSelf: "center",
    justifyContent: "center",
    borderRadius: saveButtonSpacing,
  },
  saveButtonText: {
    fontSize: 16,
    fontFamily: avenirFont(FontStyle.DemiBold),
    color: AppColors.whiteColor,
    textAlign: "center",
  }

});

export default WeeklySpendingLimit

