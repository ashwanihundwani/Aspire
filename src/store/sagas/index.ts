/**
 *  Redux saga class init
 */
import { takeEvery, all } from 'redux-saga/effects';
import * as types from '../actions/types';
import requestDebitCardAsync from './debitCardSaga';

//Watcher Saga
export default function* watch() {
  yield all([takeEvery(types.DEBIT_CARD_INFO_REQUEST, requestDebitCardAsync)]);
}
