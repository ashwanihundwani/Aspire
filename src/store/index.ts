import { createStore, compose, applyMiddleware } from 'redux';


import createSagaMiddleware from 'redux-saga';

import rootReducer from './reducers'; // where reducers is a object of reducers
import sagas from './sagas';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(sagas);

export default store