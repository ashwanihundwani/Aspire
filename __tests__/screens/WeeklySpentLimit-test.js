/**
 * @format
 */

import 'react-native';
import React from 'react';
import WeeklySpendingLimit from '../../src/screens/WeeklySpendingLimit';
import renderer from 'react-test-renderer';
import store from '../../src/store';
import { Provider } from 'react-redux';

it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><WeeklySpendingLimit /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});