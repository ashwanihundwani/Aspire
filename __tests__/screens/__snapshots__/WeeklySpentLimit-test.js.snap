// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#0C365A",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "AvenirNext-Bold",
        "fontSize": 24,
        "marginLeft": 43.5,
        "marginTop": 10,
      }
    }
  >
    Spending Limit
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
        "flex": 1,
        "height": 1204,
        "marginTop": 60,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginLeft": 43.5,
            "marginTop": 43.5,
          }
        }
      >
        <Image
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "marginRight": 12,
            }
          }
        />
        <Text>
          Set a weekly debit card spending limit
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginLeft": 43.5,
            "marginTop": 10,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#01D167",
              "borderRadius": 4,
              "height": 22,
              "marginTop": 5,
              "width": 40,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "AvenirNext-Bold",
                "fontSize": 12,
                "marginTop": 4,
                "textAlign": "center",
              }
            }
          >
            S$
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "AvenirNext-Bold",
                "fontSize": 24,
                "marginLeft": 10,
                "textAlign": "center",
              },
              Object {
                "color": "#222222",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#DDDDDD",
            "height": 0.7,
            "marginLeft": 43.5,
            "marginTop": 3,
            "width": 663,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#22222266",
            "fontFamily": "AvenirNext-Regular",
            "fontSize": 13,
            "marginLeft": 43.5,
            "marginTop": 14,
          }
        }
      >
        Here weekly means the last 7 days - not the calendar week
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginLeft": 43.5,
            "marginTop": 43.5,
            "width": 663,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(1,209,103, 0.07)",
                "borderRadius": 4,
                "height": 40,
                "justifyContent": "center",
                "width": 206.25000000000003,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#01D167",
                  "fontFamily": "AvenirNext-DemiBold",
                  "fontSize": 12,
                  "position": "absolute",
                  "textAlign": "center",
                }
              }
            >
              S$ 5,000
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(1,209,103, 0.07)",
                "borderRadius": 4,
                "height": 40,
                "justifyContent": "center",
                "width": 206.25000000000003,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#01D167",
                  "fontFamily": "AvenirNext-DemiBold",
                  "fontSize": 12,
                  "position": "absolute",
                  "textAlign": "center",
                }
              }
            >
              S$ 10,000
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(1,209,103, 0.07)",
                "borderRadius": 4,
                "height": 40,
                "justifyContent": "center",
                "width": 206.25000000000003,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#01D167",
                  "fontFamily": "AvenirNext-DemiBold",
                  "fontSize": 12,
                  "position": "absolute",
                  "textAlign": "center",
                }
              }
            >
              S$ 20,000
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "justifyContent": "flex-end",
          "marginBottom": 50,
        }
      }
    >
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#EEEEEE",
            "borderRadius": 101.25,
            "height": 101.25,
            "justifyContent": "center",
            "opacity": 1,
            "width": 547.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "AvenirNext-DemiBold",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</View>
`;
